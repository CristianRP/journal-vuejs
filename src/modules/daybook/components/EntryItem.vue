<template>
  <div class="entry-container mb-3 pointer p-2">
    <!-- Title -->
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">25</span>
      <span class="mx-1 fs-5">May</span>
      <span class="mx-2 fw-light">2022, Wed</span>
    </div>

    <div class="entry-description">
      Officia aliquip officia quis irure veniam ad ex Lorem. Officia consequat ea eu consequat et sint dolor eiusmod eu id anim culpa deserunt in. Magna culpa nisi minim aute ut eiusmod duis non exercitation ullamco magna non. Ea nulla deserunt ut dolor nisi minim ex ullamco pariatur elit. Eu ipsum exercitation laboris ullamco ut ad sit ex ea non pariatur.
    </div>
  </div>
</template>

<script>
export default {

}
</script>

<style lang="scss" scoped>
.entry-container {
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s all ease-in;

  &:hover {
    background-color: lighten($color: grey, $amount: 45);
    transition: 0.2s all ease-in;
  }

  .entry-description {
    font-size: 12px;
  }
}
</style>