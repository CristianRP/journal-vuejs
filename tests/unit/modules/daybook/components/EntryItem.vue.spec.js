import { shallowMount } from '@vue/test-utils'
import EntryItem from '@/modules/daybook/components/EntryItem.vue'

describe('Tests on EntryItem component', () => {

  let mockRouter, wrapper

  beforeEach(() => {
    // mockRouter
    mockRouter = {
      push: jest.fn()
    }

    // const wrapper = shallowMount( entry, {.. props .. global} )
    wrapper = shallowMount( EntryItem, {
      props: {
        entry: {
                  "id": "-N3PLntxWSjEgGDGZepB",
                  "date": 1654004594018,
                  "text": "Hello From Journal\n'Labore dolor laborum aliqua est excepteur labore Lorem. Laborum sint elit sit culpa labore. Eiusmod id et amet elit exercitation. Eu amet deserunt commodo voluptate et. Consequat cupidatat aliqua id eu tempor mollit laborum sunt cupidatat ipsum Lorem. Reprehenderit labore do amet proident adipisicing nisi exercitation. Consectetur do magna nulla ut eiusmod do nulla veniam officia laborum et cillum tempor.',"
              }
      },
      global: {
        mocks: {
          $router: mockRouter
        }
      }
    })
  })

  test('should match with the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should redirect when click on the container', () => {
    wrapper.find('.entry-container').trigger('click')

    expect( mockRouter.push ).toHaveBeenCalled()
    expect( mockRouter.push ).toHaveBeenCalledWith({
      name: 'entry',
      params: {
        id: expect.any(String)
      }
    })
  })

  test('tests on computed properties', () => {
    // wrapper.vm <-- see computed properties
    // day: 23
    // month: July
    // yearDay: '2022, Friday'
    expect( typeof wrapper.vm.day ).toBe("number")
    expect( typeof wrapper.vm.month ).toBe("string")
    expect( typeof wrapper.vm.yearDay ).toBe("string")

  })
})

