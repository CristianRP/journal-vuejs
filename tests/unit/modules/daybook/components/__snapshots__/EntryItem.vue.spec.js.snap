// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on EntryItem component should match with the snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- Title -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">31</span><span class="mx-1 fs-5">May</span><span class="mx-2 fw-light">2022, Thursday</span></div>
  <div class="entry-description">Hello From Journal
    'Labore dolor laborum aliqua est excepteur labore Lorem. Laborum sint elit sit culpa labore. Eiusmod id et amet...</div>
</div>
`;
